src/sentry/integrations/pagerduty/integration.py

```diff
@@ -0,0 +1,166 @@
+    aspects={},
+)
+
+
+class PagerDutyIntegration(IntegrationInstallation):
+    def get_client(self):
+        pass
+
+    def get_organization_config(self):
+        from sentry.models import Project
+
+        projects = Project.objects.filter(
+            organization_id__in=self.model.organizations.values_list("id", flat=True)
+        )
+        items = []
+        for p in projects:
+            items.append({"value": p.id, "label": p.name})
+
+        service_options = [(s["id"], s["name"]) for s in self.services.values()]
+
+        fields = [
+            {
+                "name": "project_mapping",
+                "type": "choice_mapper",
+                "label": "Map projects in Sentry to services in PagerDuty",
+                "help": "When an alert rule is triggered in a project, this mapping will let us know what service to create the incident in PagerDuty.",
+                "addButtonText": "Add Sentry Project",
+                "addDropdown": {
+                    "emptyMessage": "All projects configured",
+                    "noResultsMessage": "Could not find project",
+                    "items": items,
+                },
+                "mappedSelectors": {
+                    "service": {"choices": service_options, "placeholder": "Select a service"}
+                },
+                "columnLabels": {"service": "Service"},
+                "mappedColumnLabel": "Sentry Project",
+            }
+        ]
+
+        return fields
+
+    def update_organization_config(self, data):
+
+        if "project_mapping" in data:
+            project_ids_and_services = data.pop("project_mapping")
+            with transaction.atomic():
+                PagerDutyServiceProject.objects.filter(
+                    organization_integration_id=self.org_integration.id
+                ).delete()
+
+                for p_id, s in project_ids_and_services.items():
+                    # create the record in the table
+                    project = Project.objects.get(pk=p_id)
+                    service = self.services[s["service"]]
+                    PagerDutyServiceProject.objects.create(
+                        organization_integration=self.org_integration,
+                        project=project,
+                        service_name=service["name"],
+                        service_id=service["id"],
+                        integration_key=service["integration_key"],
+                    )
+
+    def get_config_data(self):
+        config = self.org_integration.config
+        project_mappings = PagerDutyServiceProject.objects.filter(
+            organization_integration_id=self.org_integration.id
+        )
+        data = {}
+        for pm in project_mappings:
+            data[pm.project_id] = {"service": pm.service_id}
+        config = {}
+        config["project_mapping"] = data
+        return config
+
+    @property
+    def services(self):
+        return self.model.metadata["services"]
```

Comment:
@MeredithAnya I feel a bit uncomfortable relying on metadata for core functionality. At the DB level, we have no guarantee that `services` will exist. I would much prefer to have a service table so we can make this more SQL friendly.